### PR TITLE
Thk firmwarecheck

### PR DESCRIPTION
**Pull request by TomKimsey**

### CODE REVIEW

File: src/main/java/com/hms_networks/americas/sc/extensions/system/info/SCSystemInfo.java [hunk 1]
```diff
@@ -23,6 +23,90 @@ public class SCSystemInfo {
...
+    return currFwVerMinor;
+  }
+
+  /**
+   * Gets the major firmware version as an int.
+   *
+   * @return major firmware version
+   * @throws EWException for Flexy specific Exception
+   */
+  private static int getFirmwareVerMajor() throws EWException {
```
Comment (alexjhawk) [inline src/main/java/com/hms_networks/americas/sc/extensions/system/info/SCSystemInfo.java:69]: Make public

File: src/main/java/com/hms_networks/americas/sc/extensions/system/info/SCSystemInfo.java [hunk 2]
```diff
@@ -23,6 +23,90 @@ public class SCSystemInfo {
...
+  /** Value for current required Ewon firmware major version */
+  private static int currReqFwVerMinor = DEFAULT_MIN_MINOR_FW_VER;
+
+  /**
+   * Gets the minor firmware version as an int.
+   *
+   * @return minor firmware version
+   * @throws EWException for Flexy specific Exception
+   */
+  private static int getFirmwareVerMinor() throws EWException {
```
Comment (alexjhawk) [inline src/main/java/com/hms_networks/americas/sc/extensions/system/info/SCSystemInfo.java:53]: Make public

**Review by alexjhawk — APPROVED** [review]
Looks good, although we might as well keep the firmware version methods publicly accessible. A connector implementing firmware update capabilities, like the Cumulocity Connector, could use it.

File: src/main/java/com/hms_networks/americas/sc/extensions/system/info/SCSystemInfo.java (hunk 2 above)
Comment (TomKimsey) [reply src/main/java/com/hms_networks/americas/sc/extensions/system/info/SCSystemInfo.java]: changed

File: src/main/java/com/hms_networks/americas/sc/extensions/system/info/SCSystemInfo.java (hunk 1 above)
Comment (TomKimsey) [reply src/main/java/com/hms_networks/americas/sc/extensions/system/info/SCSystemInfo.java]: changed

**Review by oliver-walker-hms — APPROVED** [review]

**Review by alexjhawk — APPROVED** [review]

**Pull request merged**